flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/PatternStream.java

```diff
@@ -107,13 +113,54 @@
 	 *         function.
 	 */
 	public <R> SingleOutputStreamOperator<R> select(final PatternSelectFunction<T, R> patternSelectFunction, TypeInformation<R> outTypeInfo) {
-		SingleOutputStreamOperator<Map<String, List<T>>> patternStream =
-				CEPOperatorUtils.createPatternStream(inputStream, pattern);
+		return CEPOperatorUtils.createPatternStream(inputStream, pattern, clean(patternSelectFunction), outTypeInfo);
+	}
+
+	/**
+	 * Applies a select function to the detected pattern sequence. For each pattern sequence the
+	 * provided {@link PatternSelectFunction} is called. The pattern select function can produce
+	 * exactly one resulting element.
+	 *
+	 * <p>Applies a timeout function to a partial pattern sequence which has timed out. For each
+	 * partial pattern sequence the provided {@link PatternTimeoutFunction} is called. The pattern
+	 * timeout function can produce exactly one resulting element.
+	 *
+	 * <p>You can get the stream of late data using
```

Comment:
Maybe something like "You can get the stream of timed-out data resulting from the {@link PatternTimeoutFunction} using ..."